iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/agent/SubscriptionConsumerAgent.java

```diff
@@ -0,0 +1,285 @@
+              .setTopicNames(topicNames);
+      final TSStatus tsStatus = configNodeClient.dropSubscription(req);
+      if (TSStatusCode.SUCCESS_STATUS.getStatusCode() != tsStatus.getCode()) {
+        final String exceptionMessage =
+            String.format(
+                "Subscription: Failed to unsubscribe topics %s for consumer %s in config node, status is %s.",
+                topicNames, consumerConfig, tsStatus);
+        LOGGER.warn(exceptionMessage);
+        throw new SubscriptionException(exceptionMessage);
+      }
+    } catch (ClientManagerException | TException e) {
+      final String exceptionMessage =
+          String.format(
+              "Subscription: Failed to unsubscribe topics %s for consumer %s in config node, exception is %s.",
+              topicNames, consumerConfig, e.getMessage());
+      LOGGER.warn(exceptionMessage);
+      throw new SubscriptionException(exceptionMessage);
+    }
+  }
+
+  ////////////////////////// ConsumerGroupMeta Lock Control //////////////////////////
+
+  protected void acquireReadLock() {
+    consumerGroupMetaKeeper.acquireReadLock();
+  }
+
+  protected void releaseReadLock() {
+    consumerGroupMetaKeeper.releaseReadLock();
+  }
+
+  protected void acquireWriteLock() {
+    consumerGroupMetaKeeper.acquireWriteLock();
+  }
+
+  protected void releaseWriteLock() {
+    consumerGroupMetaKeeper.releaseWriteLock();
+  }
+
+  ////////////////////////// ConsumerGroupMeta Management Entry //////////////////////////
+
+  public TPushConsumerGroupRespExceptionMessage handleSingleConsumerGroupMetaChanges(
+      ConsumerGroupMeta consumerGroupMetaFromCoordinator) {
+    acquireWriteLock();
+    try {
+      handleSingleConsumerGroupMetaChangesInternal(consumerGroupMetaFromCoordinator);
+      return null;
+    } catch (Exception e) {
+      final String consumerGroupId = consumerGroupMetaFromCoordinator.getConsumerGroupId();
+      final String exceptionMessage =
+          String.format(
+              "Subscription: Failed to handle single consumer group meta changes for consumer group %s, because %s",
+              consumerGroupId, e.getMessage());
+      LOGGER.warn(exceptionMessage);
+      return new TPushConsumerGroupRespExceptionMessage(
+          consumerGroupId, exceptionMessage, System.currentTimeMillis());
+    } finally {
+      releaseWriteLock();
+    }
+  }
+
+  private void handleSingleConsumerGroupMetaChangesInternal(
+      final ConsumerGroupMeta metaFromCoordinator) {
+    final String consumerGroupId = metaFromCoordinator.getConsumerGroupId();
+    if (!SubscriptionAgent.broker().isBrokerExist(consumerGroupId)) {
+      SubscriptionAgent.broker().createBroker(consumerGroupId);
+    }
+    consumerGroupMetaKeeper.removeConsumerGroupMeta(consumerGroupId);
+    consumerGroupMetaKeeper.addConsumerGroupMeta(consumerGroupId, metaFromCoordinator);
+  }
+
+  public TPushConsumerGroupRespExceptionMessage handleConsumerGroupMetaChanges(
```

Comment:
We have to check meta exists locally but not exists in consumerGroupMetasFromCoordinator